Add overrides configmap to querier, update docs for new limit (#1153)
grafana/tempo: docs/tempo/website/configuration/_index.md

    
    # Global ingestion limits configurations

>     # Specifies whether the ingestion rate limits should be applied by each replica (of the distributor and ingester)
>     # individually OR should be shared across all replicas. See the "override strategies" section below for an example.

Reviewer: Wording suggestions here are to attempt to use similar terms across all documentation.  Here, we're talking about components, so I've substituted "instance" for "replica." 

```suggestion
    # Specifies whether the ingestion rate limits should be applied by each instance of the distributor and ingester
    # individually, or the limits are to be shared across all instances. See the "override strategies" section for an example.
```

    # Maximum number of active traces per user, per ingester. `0` to disable. 
    # Results in errors like
    #    LIVE_TRACES_EXCEEDED: max live traces per tenant exceeded: per-user traces limit (local: 10000 global: 0 actual local: 1) exceeded
    # This override limit is used by the ingester.
    [max_traces_per_user: <int> | default = 10000]

    # Maximum size of search data for a single trace in bytes. `0` to disable.
    # From an operational perspective, the size of search data is proportional to the total size of all tags in a trace
    [max_search_bytes_per_trace: <int> | default = 5000]

    # Maximum size in bytes of a tag-values query. Tag-values query is used mainly to populate the autocomplete dropdown.

>     # Limit added to protect from tags with high cardinality or large values (like http urls or sql queries)

Reviewer: ```suggestion
    # Limit added to protect from tags with high cardinality or large values (like HTTP URLs or SQL queries)
```

    # This override limit is used by the ingester and the querier.
    [max_bytes_per_tag_values_query: <int> | default = 5000000 (5MB) ]

    # Tenant-specific overrides

>     # Tenant-specific overrides settings config file. See the "Tenant-specific overrides" section below for an example.

Reviewer: Note:  "config" is not a word (or an abbreviation). 
```suggestion
    # Tenant-specific overrides settings configuration file. See the "Tenant-specific overrides" section for an example.
```

A setting that applies at a local level is quite helpful in ensuring that each distributor independently can process traces up to the limit without affecting the tracing limits on other distributors.

However, as a cluster grows quite large, this can lead to quite a large quantity of traces. An alternative strategy may be to set a `global` trace limit that establishes a total budget of all traces across all distributors in the cluster. The global limit is averaged across all distributors by using the distributor ring.

```yaml
# /conf/tempo.yaml
overrides:
  [ingestion_rate_strategy: <global|local> | default = local]
```

> For ex, the config below specifies that each replica of the distributor will apply a limit of `15MB/s`.

Reviewer: ```suggestion
For example, this configuration specifies that each instance of the distributor will apply a limit of `15MB/s`.
```

```yaml
overrides:
  - ingestion_rate_strategy: local
  - ingestion_rate_limit_bytes: 15000000
```

> The config below specifies that together, all distributor replicas will apply a limit of `15MB/s`.
> So if there are 5 replicas, each will apply a local limit of `(15MB/s / 5) = 3MB/s`.

Reviewer: ```suggestion
This configuration specifies that together, all distributor instances will apply a limit of `15MB/s`.
So if there are 5 instances, each instance will apply a local limit of `(15MB/s / 5) = 3MB/s`.
```
